### PR TITLE
refactor both team and team_factory to include class methods/variables

### PR DESCRIPTION
**Pull request by austincarrjones**
## Type Of Change
- [ ] New Feature
- [ ] Debugging
- [X] Refactor/File Clean-up
- [ ] Tests
- [ ] Documentation

## Provide A Summary Of Your Change
Updated classes to use the class variables/methods structure you guys are using for game and game_teams. Also moved #count_of_teams to team class. Thanks for troubleshooting that one Zach and Noah. Also removed unnecessary attributes YAGNI.

## Group Project Check List
- [X] I made test for my code, and made sure they were passing before sucmitting PR
- [X] I assigned atleast 2 group members to review my PR

### CODE REVIEW

File: lib/team.rb [hunk 1]
```diff
@@ -1,20 +1,18 @@
 class Team
   attr_reader :team_id,
-              :franchise_id,
-              :team_name,
-              :abbreviation,
-              :stadium,
-              :link
+              :team_name
```
Comment (zach-bergman) [inline lib/team.rb:3]: YAGNI BABY - at least for now, we'll see what stats we need as we move on, might have to add more in later.

Comment (noahdurbin) [reply lib/team.rb:3]: I need to do this in the game class too!

File: lib/team_factory.rb [hunk 2]
```diff
@@ -1,24 +1,19 @@
...
-  def create_teams
-    CSV.foreach('./data/teams.csv', headers: true, header_converters: :symbol) do |row|
-      @teams << Team.new(
-      team_id: row[:team_id].to_i,
-      franchise_id: row[:franchise_id].to_i,
-      team_name: row[:teamname],
-      abbreviation: row[:abbreviation],
-      stadium: row[:stadium],
-      link: row[:link])
+  def self.create_teams(path)
```
Comment (zach-bergman) [inline lib/team_factory.rb:4]: Thanks for turning this into a Class method!

File: spec/team_spec.rb [hunk 3]
```diff
@@ -3,21 +3,20 @@
 RSpec.describe TeamFactory do
 
   it 'exists' do
-    team = Team.new({team_id: 1, franchiseid: 23, team_name: "Atlanta United", abbreviation: "ATL", stadium: "Mercedes-Benz Stadium", link: "/api/v1/teams/1"})
+    team = Team.new({team_id: 1, team_name: "Atlanta United"})
```
Comment (noahdurbin) [inline spec/team_spec.rb:6]: so much cleaner without all that unnecessary data

**Review by noahdurbin — APPROVED** [review]
looks great!

**Review by zach-bergman — APPROVED** [review]
Looks great! I will implement ::create_teams in the StatTracker now

**Pull request merged**